src/nvim/regexp_nfa.c

```diff
@@ -6064,10 +6070,17 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
       break;
 
     // Allow interrupting with CTRL-C.
-    fast_breakcheck();
+    line_breakcheck();
```

Comment:
`make unittest` also fails on neovim master branch if the `*_breakcheck()` functions directly call `os_breakcheck()`:
```diff
diff --git a/src/nvim/misc1.c b/src/nvim/misc1.c
index 6c969a4..1d15754 100644
--- a/src/nvim/misc1.c
+++ b/src/nvim/misc1.c
@@ -2712,10 +2712,11 @@ static int breakcheck_count = 0;
 
 void line_breakcheck(void)
 {
-  if (++breakcheck_count >= BREAKCHECK_SKIP) {
-    breakcheck_count = 0;
     os_breakcheck();
-  }
+  /*if (++breakcheck_count >= BREAKCHECK_SKIP) {*/
+    /*breakcheck_count = 0;*/
+    /*os_breakcheck();*/
+  /*}*/
 }
 
 /*
@@ -2723,10 +2724,11 @@ void line_breakcheck(void)
  */
 void fast_breakcheck(void)
 {
-  if (++breakcheck_count >= BREAKCHECK_SKIP * 10) {
-    breakcheck_count = 0;
     os_breakcheck();
-  }
+  /*if (++breakcheck_count >= BREAKCHECK_SKIP * 10) {*/
+    /*breakcheck_count = 0;*/
+    /*os_breakcheck();*/
+  /*}*/
 }
```
So it does not seem that this PR is the cause for the failing tests.